includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -3242,10 +3246,7 @@ private function is_platform_payment_method( bool $is_using_saved_payment_method
 		// Make sure the payment method being charged was created in the platform.
 		if (
 			! $is_using_saved_payment_method &&
-			$this->should_use_stripe_platform_on_checkout_page() &&
-			// This flag is useful to differentiate between PRB, blocks and shortcode checkout, since this endpoint is being used for all of them.
-			! empty( $_POST['wcpay-is-platform-payment-method'] ) && // phpcs:ignore WordPress.Security.NonceVerification
-			filter_var( $_POST['wcpay-is-platform-payment-method'], FILTER_VALIDATE_BOOLEAN ) // phpcs:ignore WordPress.Security.NonceVerification,WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+			$this->should_use_stripe_platform_on_checkout_page()
```

Comment:
Just adding this as a more formal way of review for visibility.

I didn't test or look into the other parts of the code but tested the PRB functionality (Google Pay) as requested, and it [throws an error](https://github.com/Automattic/woocommerce-payments/pull/4638#discussion_r981473141).